scipy/linalg/_basic.py

```diff
@@ -315,14 +315,14 @@ def solve(a, b, lower=False, overwrite_a=False,
         rcond, info = _gtcon(dl, d, du, du2, ipiv, anorm)
     # Banded case
     elif assume_a == 'banded':
-        gbsv, = get_lapack_funcs(('gbsv',), (a1, b1))
+        gbsv, gbcon = get_lapack_funcs(('gbsv', 'gbcon'), (a1, b1))
         # Next two lines copied from `solve_banded`
         a2 = np.zeros((2*n_below + n_above + 1, ab.shape[1]), dtype=gbsv.dtype)
         a2[n_below:, :] = ab
-        _, _, x, info = gbsv(n_below, n_above, a2, b1,
-                             overwrite_ab=True, overwrite_b=overwrite_b)
+        lu, piv, x, info = gbsv(n_below, n_above, a2, b1,
```

Comment:
This needs to go in to `?gbtrf` first then it is suitable for `?gbcon` but why segfaults I can't see right away. `?gbsv` gives you the solution by combining the `?gbtrf` and `?gbtrs` steps.

Reply:
Hmm I guess not, for some reason we wrapped the output to be the factorization apparently.

Reply:
I tried ?gbtrf but it still segfaulted so I'm guessing the problem originates from ?gbcon

Reply:
It is probably the `piv` which is being zero indexed in the `gbsv` wrapper. Could you try adding 1 to `piv` before sending it to `gbcon` to make it 1-indexed?

Reply:
Indeed you are absolutely correct, that fixed the issue. Is there anything in the wrapper I should change or is this an oddity we have to live with?

Reply:
Looks like I should add a `for(i=0;i\<n;++ipiv[i++]);` to gbcon for consistency with the rest of the banded functions

Reply:
Whichever seems consistent to your eyes, I'm OK with it. Probably you have a fresher look on them as a whole. I'm a bit blinded after looking at them for a long time.